src/dtx/dtx_coll.c

```diff
@@ -106,6 +107,43 @@ dtx_coll_prep_ult(void *arg)
 	D_ASSERT(rc == ABT_SUCCESS);
 }
 
+static void
+dtx_coll_find_rank_range(struct pl_map *map, struct pl_obj_layout *layout, uint32_t version,
+			 uint32_t *min_rank, uint32_t *max_rank)
+{
+	struct pool_target *target;
+	d_rank_t            my_rank = dss_self_rank();
+	int                 rc;
+	int                 i;
+
+	for (i = 0, *min_rank = -1, *max_rank = 0; i < layout->ol_nr; i++) {
+		if (layout->ol_shards[i].po_target == -1 || layout->ol_shards[i].po_shard == -1)
+			continue;
+
+		rc = pool_map_find_target(map->pl_poolmap, layout->ol_shards[i].po_target, &target);
+		D_ASSERT(rc == 1);
+
+		/* Skip current leader rank. */
+		if (target->ta_comp.co_rank == my_rank)
+			continue;
+
+		/* Skip the target that (re-)joined the system after the DTX. */
+		if (target->ta_comp.co_ver > version)
```

Comment:
seems the co_ver only set when it initially added to pool map? when it be excluded+reinted seems only set co_in_ver.
in DTX code some places use co_ver, some other places use co_in_ver, probably need to unify them in a following PR.

And this comparison looks better to be after below L135 ~L138?

Reply:
> seems the co_ver only set when it initially added to pool map? when it be excluded+reinted seems only set co_in_ver.
> in DTX code some places use co_ver, some other places use co_in_ver, probably need to unify them in a following PR.

Yes, use co_in_ver may filter out more unnecessary ranks, that is better.

> And this comparison looks better to be after below L135 ~L138?

Changing the comparison order will get the same result.